test/e2e/nix/build_test.go

```diff
@@ -0,0 +1,55 @@
+package nix_test
```

Comment:
* Add a test with ambiguous dependencies defined in root.
* Add a test with ambiguous dependencies defined on the task dependency list.
```
dependencies: [
   go_1_18
   go_1_17
]
```

It should have a defined behaviour which one is used. First or Last? Pick one!